src/test/test-namespace.c

```diff
@@ -102,7 +106,61 @@ static int test_netns(void) {
         n += si.si_status;
 
         assert_se(n == 1);
-        return EXIT_SUCCESS;
+}
+
+static void test_protect_kernel_logs(void) {
+        int r;
+        pid_t pid;
+        static const NamespaceInfo ns_info = {
+                .protect_kernel_logs = true,
+        };
+
+        if (geteuid() > 0) {
+                (void) log_tests_skipped("not root");
+                return;
+        }
+
+        /* In a container we likely don't have access to /dev/kmsg */
+        if (detect_container() > 0) {
+                (void) log_tests_skipped("in container");
+                return;
+        }
+
+
+        pid = fork();
+        assert_se(pid >= 0);
+
+        if (pid == 0) {
+                _cleanup_close_ int fd = -1;
+
+                fd = open("/dev/kmsg", O_RDONLY);
```

Comment:
doesn#t matter much in this case, but we made it a habit to always specifiy O_CLOEXEC unless there's explicit reason not to, so that grepping over all open()s tells us quickly that we didn't forget it anwhere.